prost-derive/src/lib.rs

```diff
@@ -7,8 +7,7 @@ extern crate proc_macro;
 
 use anyhow::{bail, Error};
 use itertools::Itertools;
-use proc_macro::TokenStream;
-use proc_macro2::Span;
+use proc_macro2::{Span, TokenStream};
```

Comment:
Why is `proc_macro2::TokenStream` better?

Reply:
`proc_macro` can't be used in tests; this seems to be the main usecase for `proc_macro2`. we translate at the boundary of the `proc_macro` api itself rather than letting its untestable types penetrate all the way into our functionality.

`quote!` unconditionally produces `proc_macro2::TokenStream` as its output; the only thing that really changes with this is that the boundary at which we translate from and back into the un-testable `proc_macro::TokenStream` type changes from *just inside* the function (from `syn::parse(input)` at the top, and via `expanded.into()` at the bottom) to just *outside* the function (where the trivial `.into()` calls happen inside the function marked with `#[proc_macro_derive]`).